share/availability/full/utils.go

```diff
@@ -0,0 +1,46 @@
+package full
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+)
+
+var (
+	storePrefix     = datastore.NewKey("full_avail")
+	previousModeKey = datastore.NewKey("previous_run")
+	pruned          = []byte("pruned")
+	archival        = []byte("archival")
+)
+
+// ConvertFromArchivalToPruned ensures that a node has not been run with pruning enabled before
+// cannot revert to archival mode. It returns true only if the node is converting to
+// pruned mode for the first time.
+func ConvertFromArchivalToPruned(ctx context.Context, ds datastore.Datastore, isArchival bool) (bool, error) {
```

Comment:
(nit) Would it be better to name like updatePruneMode, since it handles both archival -> pruned and pruned -> arhival cases now?

Reply:
I think `Convert` is more descriptive here, so I will leave it